app/controllers/users.js

```diff
@@ -107,32 +112,111 @@ exports.create = function(req, res) {
   }
 };
 
+exports.signUp = function (req, res) {
+  const {
+    name,
+    password,
+    email
+  } = req.body;
+
+  if (name && password && email) {
+    User.findOne({
+      email
+    })
+      .then((user) => {
+        if (!user) {
+          const newUser = new User(req.body);
+          newUser.avatar = avatars[newUser.avatar];
+          newUser.provider = 'local';
+          newUser.save((error, nuUser) => {
```

Comment:
Just out of curiosity guys. Please, where is `nuUser` coming from and going to? 🤔 

@clintfidel @OlatunjiYso @ddouglasz